python/tutorials/10-block-scaled-matmul.py

```diff
@@ -0,0 +1,335 @@
+"""
+Block Scaled Matrix Multiplication
+==================================
+This tutorial demonstrates a Triton implementation of block scaled matrix multiplication
+which is generic over FP4 and FP8 formats. The formats supported in the tutorial are the OCP microscaling
+formats, including mxfp4 and mxfp8, as well as NVIDIA's nvfp4 format. These matrix multiplications
+are accelerated by fifth generation tensor core instructions on CUDA devices with compute capability 10.
+
+Users can run the tutorial with each of the supported formats by passing the `--format`
+argument and can benchmark the performance of each by specifying matrix dimensions
+and iteration steps.
+
+.. code-block:: bash
+
+    # FP4
+    python 10-block-scaled-matmul.py --format nvfp4
+    python 10-block-scaled-matmul.py --format mxfp4 --K_range 512 8192 --bench
+
+    # FP8
+    python 10-block-scaled-matmul.py --format mxfp8 --K_range 8192 16384 --K_step 2048 --bench
+
+Future updates to this tutorial which support mixed precision block scaled matmul are planned.
+"""
+
+# %%
+# Background
+# ----------
+#
+# CUDA devices that support PTX 8.7 and later can utlize block scaled matrix multiply
+# instructions. In order for low latency access to these scale factors in the fast
+# inner loop over tensor core MMAs, it is important to ensure that the blocked
+# scale factors are stored in a contiguous memory layout according to their access
+# pattern.
+#
+# The block scaled matmul tensor core instructions compute the following product:
+#
+#     C = (A * scale_a) @ (B * scale_b)
+#
+# where scale_a and scale_b are the blocked scale factors for the A and B matrices.
+# Under block scaled matmul, each scale factor is broadcast and multiplied across a
+# vector of elements from the A and B matrices, usually along their respective K axes.
+# The number of elements of A and B over which each scale factor is broadcast is herein
+# refered to as the vector size (VEC_SIZE).
+#
+# In a linear row-major layout, the scale factors would take the shape
+#
+#     (M, K // VEC_SIZE) and (N, K // VEC_SIZE)   [1]
+#
+# in global memory. However, to avoid non-contiguous memory access, it is beneficial to
+# instead store the scale factors in a packed block layout. For the LHS matrix this layout
+# is given by
+#
+#     (M // 32 // 4, K // VEC_SIZE // 4, 32, 4, 4)   [2].
+#
+# In this way, each tensor core MMA in the fast inner loop over K blocks can achieve contiguous
+# access of a block of 128 rows of scale factors along the M axis, for each BLOCK_M x BLOCK_K
+# subtile of the matrix A.
+#
+# In order to conform with Triton's language semantics for dot_scaled, the scale factors
+# are prepared in the above 5D layout [2], but are then logically transposed and reshaped into
```

Comment:
Are the reference [2] on tcgen.cp here and L53 relevant enough? I'd rather consider tcgen05.cp an implementation detail that users do not need to care about. Simply having this layout, but without using TMEM copy, is till much faster than the naive layout.

We may consider offloading the discussion on global memory layout to cublas or cutlass doc:
https://docs.nvidia.com/cuda/cublas/#d-block-scaling-factors-layout
https://github.com/NVIDIA/cutlass/blob/main/media/docs/blackwell_functionality.md#scale-factor-layouts